cache/cache_rep.cc

```diff
@@ -0,0 +1,35 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+#include <cassert>
+#include <cstddef>
+#include <cstring>
+#include <memory>
+
+#include "cache/cache_rep.h"
+#include "rocksdb/cache.h"
+#include "rocksdb/slice.h"
+#include "table/block_based/block_based_table_reader.h"
+#include "util/coding.h"
+
+namespace ROCKSDB_NAMESPACE {
+CacheRep::CacheRep(std::shared_ptr<Cache> cache)
+    : cache_allocated_size_(0) {
+  assert(cache != nullptr);
```

Comment:
[minor_improvement]
I think it is safe to use "assert" here for an extra check because the CacheRep class users should pass in a valid Cache pointer if they want a CacheRep on it. I also double checked that the only user of this CacheRep class - WBM - did null check against the cache before passing it into the constructor.